pkg/workflows/wasm/host/module.go

```diff
@@ -24,31 +25,42 @@ import (
 	wasmpb "github.com/smartcontractkit/chainlink-common/pkg/workflows/wasm/pb"
 )
 
-type respStore struct {
-	m  map[string]*wasmpb.Response
+type RequestData struct {
+	response *wasmpb.Response
+	ctx      context.Context
+}
+
+type store struct {
+	m  map[string]*RequestData
 	mu sync.RWMutex
 }
 
-func (r *respStore) add(id string, resp *wasmpb.Response) error {
+func (r *store) add(id string, req *RequestData) error {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	_, found := r.m[id]
-	if found {
+	storedReq, found := r.m[id]
+	if found && req.response != nil {
 		return fmt.Errorf("error storing response: response already exists for id: %s", id)
 	}
 
-	r.m[id] = resp
+	// we only add the response as the context has already been added
+	if found && req.response == nil {
+		r.m[id].response = storedReq.response
+		return nil
+	}
+
+	r.m[id] = req
```

Comment:
@agparadiso I think you're right: we can just let this be the same as before (i.e. revert the diff to this method), and assume we'll only add it once

If we want to mutate the entry, we can get the entry and modify it 👍